### ISSUE TITLE
Add support for 7z file type

### ISSUE THREAD

**Issue by uBadRequest** [phiresky/ripgrep-all#72]
None of the adapters seem to support 7z, can't you just use 7z e -oc ?

I couldn't figure out how to get it to work with ripgrep's preprocessor

**Comment by phiresky** [comment]
Not really, since files within archives need to be handled individually - so this needs a custom implementation for 7z. Haven't looked at rust libraries for 7z yet.
